### DISCUSSION TITLE
PS, top, vmstat commands not found in Strimzi Kafka broker pods

### DISCUSSION THREAD

**Discussion by Sai-Charan-Madhvaraj** [strimzi/strimzi-kafka-operator#7122] [Ideas]
Team,

When we are trying to find the process, cpu and memory stats inside the Kafka broker...none of the commands are present.

eg: top, free, vmstat, sar

Strimzi Version : 0.27.1
kafka version : 3.0.0

We might need to have these packages included in base images so that will be useful for any troubleshooting purposes.

**Comment by scholzj** [comment]
If you want, you can always extend the image to contain additional packages. I'm not sure why do you need some of these commands (e.g. what is the value of `top` inside a container?). 

But in general:
* More tools means bigger container images
* More tools means bigger footprint for CVEs

So there will be always some users arguing for having as few tools installed as possible and some users arguiing for more tools based on what they are used to.

**Reply by Sai-Charan-Madhvaraj** [reply]
Yeah, completely agree ..that having more packages will be vulnerable and size grows.

but how about adding **ps** command which is basically required eg:  list of process and command line args of the JVM for any troubleshooting purposes.

**Reply by scholzj** [reply]
It always starts with `I completely agree ... but what about the xyz utility which is surely needed` :-D ... the command of the running application can be lister through the `/proc` directory structure. That is what for example our tests use to check what was started.
